nowcasting_metrics/metrics/mae.py

```diff
@@ -118,89 +124,117 @@ def make_pvlive_mae(
     return value, number_of_data_points
 
 
-def make_mae_one_gsp_with_forecast_horizon(
+def make_mae_values(
     session: Session,
     datetime_interval: DatetimeInterval,
-    gsp_id: int,
-    forecast_horizon_minutes: int,
-    model: Optional[Union[ForecastValueSQL, ForecastValueSevenDaysSQL]] = None,
+    forecast_values: pd.DataFrame,
+    gsp_yields: pd.DataFrame,
     metric: Optional[Metric] = latest_mae,
     model_name: Optional[str] = None,
+    use_adjuster: bool = True,
+    forecast_horizon_minutes: Optional[int] = None
 ) -> (int, int):
     """
-    Calculate the MAE for one GSP for a forecast horizon, and save to database
+    Calculate the MAE for one GSP, and save to database
 
     :param session: database session
-    :param datetime_interval: datetime interbal
-    :param gsp_id: the gsp id
-    :param forecast_horizon_minutes: the forecast horizon ie. Use results from forecast that are
-        made 60 minutes before target time
-    :param model: the model to use
+    :param datetime_interval: datetime interval
+    :param use_adjuster: option to use the adjuster or not.
     :param metric: the metric to use
     :param model_name: the model name of the forecast. This is optional.
-    :return: 1. the MAE, 2. the MAE with the adjuster 3. the number of data points
+    :param use_adjuster: option to use the adjuster or not.
+    :param forecast_horizon_minutes: the forecast horizon ie. Use results from forecast that are
+    made 60 minutes before target time
+    :return: 1. the MAE, 2. MAE with adjuster, 3. the number of data points
     """
 
     logger.debug(
-        f"Calculating {metric.name} for {gsp_id=} "
-        f"and {forecast_horizon_minutes=} for {model_name=}"
-    )
-
-    if model is None:
-        model = ForecastValueSevenDaysSQL
-
-    sub_query_gsp = make_gsp_sub_query(datetime_interval, gsp_id, session)
-    sub_query_forecast = make_forecast_sub_query(
-        datetime_interval,
-        forecast_horizon_minutes,
-        gsp_id,
-        session,
-        model=model,
-        model_name=model_name,
+        f"Calculating MAE for last forecast for {model_name=} for"
+        f"start={datetime_interval.start_datetime_utc} "
+        f"and end-{datetime_interval.end_datetime_utc}"
+        f" and {forecast_horizon_minutes=}"
     )
 
-    # make full query
-    query = make_mae_query(session, model=model)
-
-    query = query.filter(model.uuid.in_(sub_query_forecast))
-    query = query.filter(GSPYieldSQL.id.in_(sub_query_gsp))
-    query = query.filter(GSPYieldSQL.datetime_utc == model.target_time)
-    results = query.all()
+    if len(forecast_values) == 0:
+        logger.warning(
+            f"Forecast values are empty for {model_name=}"
+        )
+        return []
```

Comment:
The return type of this function seems to be a tuple of ints, not an array

Reply:
thanks